### PR TITLE
Prep for release v2.1.0 and add proxy config & disabling TLS examples to README

### PR DESCRIPTION
**Pull request by rohanshah18**
## Problem

- Prepare for releasing v2.1.0.
- Added support proxy configuration and for disabling TLS, so add examples to the README.

## Solution

- Update pinecone client version to `v2.1.0`.
- Add proxy config and disabling TLS examples to README

## Type of Change

- [X] Non-code change (docs, etc)

## Test Plan

NA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Maven:
 
 Gradle:
 ```
-implementation "io.pinecone:pinecone-client:1.2.2"
+implementation "io.pinecone:pinecone-client:2.0.0"
```
Comment (austin-denoble) [inline README.md:26]: Should this be `2.1.0`?

File: README.md [hunk 2]
```diff
@@ -152,7 +220,6 @@ IndexModel indexModel = pinecone.createPodsIndex(indexName, dimension, environme
 
 
 ## List indexes
-
 The following example returns all indexes (and their corresponding metadata) in your project.
```
Comment (austin-denoble) [inline README.md:223]: **nit:** I think it's generally considered best practice leave line breaks on either side of headings in markdown. My IDE has some formatting that automatically applies these when I'm working in these files: https://www.markdownguide.org/basic-syntax/#heading-best-practices

**Review by austin-denoble — APPROVED** [review]
LGTM, seems like there's a version in an example in the README which may need to be tweaked.

File: README.md (hunk 2 above)
Comment (rohanshah18) [reply README.md:223]: Ahh interesting, let me add new lines

File: README.md (hunk 1 above)
Comment (rohanshah18) [reply README.md:26]: So this was a typo from the last time and going forward, I'm planning to follow python SDKs format where we only update the versions in the README for major releases. Also I'll be adding a CI job to automate publishing.

**Pull request merged**